go.mod

```diff
@@ -1,4 +1,27 @@
-module github.com/gocql/gocql
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+// Content before git sha 34fdeebefcbf183ed7f916f931aa0586fdaa1b40
+// Copyright (c) 2016, The Gocql authors,
+// provided under the BSD-3-Clause License.
+// See the NOTICE file distributed with this work for additional information.
+//
+module github.com/apache/cassandra-gocql-driver
```

Comment:
Changing the module path seems like a backwards-incompatible change to me. 

If we change the module path, users will be able to use `github.com/apache/cassandra-gocql/driver` import path and go.mod entry to include the driver in their code. However, existing code libraries (like scylladb/gocqlx) that depend on gocql would still depend on the original gocql and not the new one. It seems to me that it will not be possible to use a [replace](https://go.dev/ref/mod#go-mod-file-replace) directive to switch to the new version either, because

> Regardless of whether a replacement is specified with a local path or module path, if the replacement module has a go.mod file, its module directive must match the module path it replaces.

On the other hand, the new module path would allow us to remove deprecated stuff as we don't need to keep the backwards-compatibility.

If we don't change the module path, all users of the library would probably have to use the replace directive in go.mod, which is not ideal either. Not sure how it works if redirects are involved.

Reply:
Thanks for catching this.

I didn't realise that changing these coordinates would be so impactful.  I think it wise to undo this change for now.   The project can discuss whether it wants to do it at a later point, e.g. in a new major version.

Reply:
In that case, all the import paths in `.go` files may need to stay as `github.com/gocql/gocql` as well.

I don't have experience moving a public Go repo, we should verify everything will work as expected on some testing repository before we setup the redirect on the main gocql repository.

Reply:
Removing the imports 👍 

But what about the `Installation` in the README.md …?
```
    go get github.com/apache/cassandra-gocql-driver
```

Reply:
We might need to do some research on how best to approach this.

It depends on what the redirect would achieve. If using the current module path works with the redirect, installation would probably work with `go get github.com/gocql/gocql`. I don't know if it would work with `go get github.com/apache/cassandra-gocql-driver`.

As for the API documentation, it would probably stay at https://pkg.go.dev/github.com/gocql/gocql too. It seems that pkg.go.dev uses the import path, so for example the documentation from kiwicom/gocql is not available there.

Reply:
For docs there's both https://godoc.org/github.com/apache/cassandra-gocql-driver and https://pkg.go.dev/github.com/apache/cassandra-gocql-driver

I'll restore them both back to `gocql/gocql` for now.